wiseconnect: Fixed setting of si91x bus events in the driver path

ragurram26 opened the pull request:
Corrected the assignments of SL_SI91X_SOCKET_DATA_TX_PENDING_EVENT,
SL_SI91X_SOCKET_COMMAND_TX_PENDING_EVENT, and
SL_SI91X_TA_BUFFER_FULL_CLEAR_EVENT to si91x_bus_event instead of
si91x_event, ensuring proper setting for the bus events.

jerome-pouiller commented:
Can you relocate the description of your PR in the commit log (the commit log is saved in git, PR descriptions are lost)

Let's try to be consistent. Wiseconnect tends to use `si91x` to name the WiFi block. So, replace `si91` by `si91x` in your commit title.

Your `Signed-off-by:` is missing.

I think it's time to adopt the `Upstream-status:` convention. If you expect this patch will be applied in the upstream, add `Upstream-Status: pending`, otherwise add `Upstream-Status: inappropriate (explanation)`.

This PR won't be merged until you send a PR to the Zephyr maintree upgrading the pointer to hal_silabs.

ragurram26 commented:
I have updated the description and title, added the signed-off, and updated the hal_silabs pointer in the west.yml file in the upstream Zephyr repo PR : https://github.com/zephyrproject-rtos/zephyr/pull/87673

jhedberg commented:
Regarding the trailers, I believe they need to be in the last paragraph, i.e. no empty line before the signed-off-by line. At least my commit message editor highlights them a special way then, but not if the trailer is not part of the last paragraph (it's not considered a trailer then)
